### DISCUSSION TITLE
Portforwarding to the App

### DISCUSSION THREAD

**Discussion by FBR65** [langgenius/dify#6837] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Hi, 

the public URL for my App is forwarded without the port.

Is there a way to set it in the .env or somewhere else?

Thanx

**Comment by akramer-zibra** [comment]
Depends on your setup @FBR65. If you use a docker setup with a compose file, you may also use a `.env` configuration file. There you configure the urls WITH the port. For example `http://.....:8080`. 

https://github.com/langgenius/dify/blob/56b43f62d1a7d677fa5d1200efbc43f0535a3f1a/docker/.env.example#L9-L37

Hope this helps. cheers

**Reply by FBR65** [reply]
Hi,

thanks for your answer. 

This doesn't work for me. I tried http://0.0.0.0:8890 but then I can't sign in any more.

On the other hand I can't set it to a IP or domain because inside the house I have an IP and from outside I have an FQDN.

**Comment by akramer-zibra** [comment]
@FBR65 If you host something on your local computer use either `http://localhost` or `http://127.0.0.1`. `0.0.0.0` is a broadcast adress and does not work to adress a single machine.

To run dify locally you may use different local domains, so nginx is able to route your requests to the correct services.

Edit your local `hosts` file to define domains for each service. Those domains will only work on your local computer, but this is enough for local hosting. There you might define some local domains like `127.0.0.1 console-api.localhost` for each service. 

This is a howto for windows computers, but you will find howtos for other operating systems in the internet aswell:
https://www.howtogeek.com/784196/how-to-edit-the-hosts-file-on-windows-10-or-11/

**Reply by FBR65** [reply]
@akramer-zibra 
Thanks for the quick response.

I thought there was another quick solution. The method shown doesn't help me because the house/public cloud architecture doesn't allow it.

**Comment by akramer-zibra** [comment]
@FBR65 you're welcome. Maybe you could use a tool like `localtunnel`: https://github.com/localtunnel/localtunnel

This tool creates a public proxy for local adresses. With this you can workaround dns issues. cheers
